warehouse/bigquery/bigquery.go

```diff
@@ -241,33 +250,142 @@ func (bq *HandleT) loadTable(tableName string, forceLoad bool, getLoadFileLocFro
 	gcsRef.MaxBadRecords = 0
 	gcsRef.IgnoreUnknownValues = false
 
-	partitionDate = time.Now().Format("2006-01-02")
-	outputTable := partitionedTable(tableName, partitionDate)
+	loadTableByAppend := func() (err error) {
+		partitionDate = time.Now().Format("2006-01-02")
+		outputTable := partitionedTable(tableName, partitionDate)
 
-	// create partitioned table in format tableName$20191221
-	loader := bq.Db.Dataset(bq.Namespace).Table(outputTable).LoaderFrom(gcsRef)
+		// create partitioned table in format tableName$20191221
+		loader := bq.Db.Dataset(bq.Namespace).Table(outputTable).LoaderFrom(gcsRef)
 
-	job, err := loader.Run(bq.BQContext)
-	if err != nil {
-		pkgLogger.Errorf("BQ: Error initiating load job: %v\n", err)
+		job, err := loader.Run(bq.BQContext)
+		if err != nil {
+			pkgLogger.Errorf("BQ: Error initiating append load job: %v\n", err)
+			return
+		}
+		status, err := job.Wait(bq.BQContext)
+		if err != nil {
+			pkgLogger.Errorf("BQ: Error running append load job: %v\n", err)
+			return
+		}
+
+		if status.Err() != nil {
+			return status.Err()
+		}
 		return
 	}
-	status, err := job.Wait(bq.BQContext)
-	if err != nil {
-		pkgLogger.Errorf("BQ: Error running load job: %v\n", err)
+
+	loadTableByMerge := func() (err error) {
+		stagingTableName = misc.TruncateStr(fmt.Sprintf(`%s%s_%s`, stagingTablePrefix, strings.ReplaceAll(uuid.Must(uuid.NewV4()).String(), "-", ""), tableName), 127)
+		pkgLogger.Infof("BQ: Loading data into temporary table: %s in bigquery dataset: %s in project: %s", stagingTableName, bq.Namespace, bq.ProjectID)
+		stagingTableColMap := bq.Uploader.GetTableSchemaInWarehouse(tableName)
+		sampleSchema := getTableSchema(stagingTableColMap)
+		metaData := &bigquery.TableMetadata{
+			Schema:           sampleSchema,
+			TimePartitioning: &bigquery.TimePartitioning{},
+		}
+		tableRef := bq.Db.Dataset(bq.Namespace).Table(tableName)
+		err = tableRef.Create(bq.BQContext, metaData)
+		if err != nil {
+			pkgLogger.Infof("BQ: Error creating temporary staging table %s", stagingTableName)
+			return
+		}
+
+		loader := bq.Db.Dataset(bq.Namespace).Table(stagingTableName).LoaderFrom(gcsRef)
+		job, err := loader.Run(bq.BQContext)
+		if err != nil {
+			pkgLogger.Errorf("BQ: Error initiating staging table load job: %v\n", err)
+			return
+		}
+		status, err := job.Wait(bq.BQContext)
+		if err != nil {
+			pkgLogger.Errorf("BQ: Error running staging table load job: %v\n", err)
+			return
+		}
+
+		if status.Err() != nil {
+			return status.Err()
+		}
+
+		if !skipTempTableDelete {
```

Comment:
any reason why we need this configurations? to avoid requiring more permissions from user? can you put it as a comment above